flake8_django/checkers/model_content_order.py

```diff
@@ -0,0 +1,107 @@
+from ast import Assign, Call, ClassDef, FunctionDef
+
+from .base_model_checker import BaseModelChecker
+from .issue import Issue
+
+
+class DJ09(Issue):
+    code = 'DJ09'
+    description = 'Order of model inner classes and standard methods does not follow Django style guide: {elem_type} should come before {before}'
+
+    def __init__(self, elem, elem_type, before):
+        super().__init__(elem.lineno, elem.col_offset, None)
+        self.description = self.description.format(
+            elem_type=elem_type,
+            before=before,
+        )
+
+
+def is_assignment_call(node):
+    # check if assigning the return value of a function call to a target which is not in all uppercase letters
+    # (because that would be a constant)
+    return isinstance(node, Assign) and isinstance(node.value, Call) and not node.targets[0].id.isupper()
+
+
+def is_manager_declaration(node):
+    return isinstance(node, Assign) and node.targets[0].id == 'objects'
+
+
+def is_meta_declaration(node):
+    return isinstance(node, ClassDef) and node.name == 'Meta'
+
+
+def is_save_declaration(node):
+    return isinstance(node, FunctionDef) and node.name == 'save'
+
+
+def is_str_declaration(node):
+    return isinstance(node, FunctionDef) and node.name == '__str__'
+
+
+def is_url_declaration(node):
+    return isinstance(node, FunctionDef) and node.name == 'get_absolute_url'
+
+
+class ModelContentOrderChecker(BaseModelChecker):
+    model_name_lookup = 'Model'
+
+    def checker_applies(self, node):
+        for base in node.bases:
+            if self.is_model_name_lookup(base) or self.is_models_name_lookup_attribute(base):
+                return True
+        return False
+
+    def get_elem_linenos(self, node):
+        # expected order of each element type
+        order = [
+            'field declaration',
+            'manager declaration',
+            'Meta class',
+            '__str__ method',
+            'save method',
+            'get_absolute_url method',
+            'custom method',
+        ]
```

Comment:
Let's store this on a class attribute so that we can re-use them. Also let's store the order on a separate dictionary, we use a dictionary in this case so that we can access the order of an element with constant time O(1)

```
    FIELD_DECLARATION = 'field declaration'
    MANAGER_DECLARATION = 'manager declaration'
    META_CLASS = 'Meta class'
    STR_METHOD = '__str__ method'
    SAVE_METHOD = 'save method'
    GET_ABSOLUTE_URL_METHOD = 'get_absolute_url method'
    CUSTOM_METHOD = 'custom method'

    MODEL_CONTENT_TYPE_EXPECTED_ORDER = {
        FIELD_DECLARATION: 0,
        MANAGER_DECLARATION: 1,
        META_CLASS: 2,
        STR_METHOD: 3,
        SAVE_METHOD: 4,
        GET_ABSOLUTE_URL_METHOD: 5,
        CUSTOM_METHOD: 6,
    }
```